### PR TITLE
Bug Fix - #129

### PR DESCRIPTION
**Pull request by vikaskandari**

### CODE REVIEW

**Comment by vikaskandari** [comment]
We were actually sending blank response even if controller method is not returning any value or undefined which was causing this issue, 

when we were using Next function in controller instead of returning any value server was sending blank response and also calling other controllers as we were using next function, but in other controller when we returned a value it was not sending any response because we have a headers already sent check which prevented it form throwing any error or any response.

To fix this i simply added a  undefined validation check if controller does not return anything or returns undefined it will not send any response , 

this way if controller uses a next function it will pass the request to next controller  and if controller is not using next function it will not return any response thus hanging response till timeout.

**Comment by vikaskandari** [comment]
Added a fix on second commit  - When server returns a undefined promise for example if we use a async method like 

```typescript
@Get("/")
async getTest(@Next() next : NextFunction){
 next();
}
```

routeHandler receives promise in result so i have set a validation if promise resolves as undefined it will not send any response

**Comment by serhiisol** [comment]
Could you revert di package.lock ?

File: express/src/express.ts [hunk 1]
```diff
@@ -79,24 +79,25 @@ function routeHandler(controller: ExpressClass, methodName: string, params: Para
     const args = extractParameters(req, res, next, params);
     const result = controller[methodName].call(controller, ...args);
 
-    if (result instanceof Promise) {
-      result.then((r: any) => {
+    if (typeof result !== "undefined") {
```
Comment (serhiisol) [inline express/src/express.ts:82]: could you add 
```typescript
if (typeof result === 'undefined') {
  return;
}
```

statement instead to reduce nesting?

File: express/src/express.ts [hunk 2]
```diff
@@ -79,24 +79,25 @@ function routeHandler(controller: ExpressClass, methodName: string, params: Para
...
+      if (result instanceof Promise) {
+        result.then((r: any) => {
+          if (!res.headersSent && typeof r !== "undefined") {
```
Comment (serhiisol) [inline express/src/express.ts:85]: Could you replace double quotes with single?

**Review by serhiisol — CHANGES_REQUESTED** [review]

**Pull request closed without merging**